ui/src/components/search/search-input.ts

```diff
@@ -0,0 +1,706 @@
+        this.selectNgmLayer(categorized.item);
+        break;
+    }
+    if (!options.keepFocus) {
+      this.inputRef.value?.blur();
+    }
+  }
+
+  private selectLocation(item: FeatureWithLocation | CoordinateItem): void {
+    if (this.viewer == null) {
+      return;
+    }
+    NavToolsStore.hideTargetPoint();
+    const rectangle = Rectangle.fromDegrees(...item.bbox);
+    if (rectangle.width < CesiumMath.EPSILON3 || rectangle.height < CesiumMath.EPSILON3) {
+      // rectangle is too small
+      const center = Rectangle.center(rectangle);
+      center.height = 5000;
+      this.viewer.camera.flyTo({
+        destination: Cartographic.toCartesian(center),
+      });
+    } else {
+      this.viewer.camera.flyTo({
+        destination: rectangle,
+      });
+    }
+  }
+
+  private selectGeoadminLayer(feature: Feature): void {
+    if (this.sidebar == null) {
+      return;
+    }
+    this.sidebar.addLayerFromSearch(feature.properties as SearchLayer).then();
+  }
+
+  private selectNgmLayer(item: EntityItem | LayerTreeNode): void {
+    if (this.sidebar == null) {
+      return;
+    }
+    NavToolsStore.hideTargetPoint();
+    const layer = isEntityItem(item) ? item : item as SearchLayerWithLayer;
+    this.sidebar.addLayerFromSearch(layer).then();
+    if (this.viewer != null && isEntityItem(item)) {
+      this.viewer.zoomTo(item.entity).then();
+    }
+  }
+
+  /**
+   * Callback that is invoked when a key is pressed while the input element is focussed.
+   *
+   * @param event The event that caused the function to be called.
+   * @private
+   */
+  private handleInputKeyPress(event: KeyboardEvent): void {
+    switch (event.key) {
+      case 'Enter': {
+        const target = this.findEnterTarget();
+        if (target != null) {
+          const [_item, element] = target;
+          element.click();
+        }
+        break;
+      }
+      case 'ArrowUp':
+      case 'ArrowDown': {
+        const target = this.findActiveResult();
+        if (target != null) {
+          const [item, _element] = target;
+          this.selectItem(item, {keepFocus: true});
+        }
+        break;
+      }
+      default:
+        break;
+    }
+  }
+
+  private findEnterTarget(): [SearchItem, HTMLLIElement] | null {
+    const {value: search} = this.searchRef;
+    const {value: results} = this.resultsRef;
+    if (search == null || results == null) {
+      return null;
+    }
+
+    // Check if there is an element that has been selected via the arrow keys.
+    const activeResult = this.findActiveResult();
+    if (activeResult != null) {
+      return activeResult;
+    }
+
+    // If there is no selected element, we attempt to select the overall first element.
+    const firstItemElement = results.children.item(0) as HTMLLIElement | null;
+    if (firstItemElement != null) {
+      return [this.resultItems[0], firstItemElement];
+    }
+    return null;
+  }
+
+  private findActiveResult(): [SearchItem, HTMLLIElement] | null {
```

Comment:
I am guessing there is something off here because the wrong serach result is zoomed to/added the catalogue (see general comment)